frontend/app/routes/$lang+/_protected+/home.tsx

```diff
@@ -35,8 +36,10 @@ export async function loader({ request }: LoaderFunctionArgs) {
   const userId = await userService.getUserId();
   const userInfo = await userService.getUserInfo(userId);
 
+  const userOrigin = await getUserOrigin(request);
+
   if (!userInfo) {
-    return redirectWithLocale(request, '/data-unavailable');
+    return redirectWithLocale(request, '/data-unavailable', { headers: { 'Set-Cookie': await userOriginCookie.serialize(userOrigin) } });
```

Comment:
Since we no longer use the `userOrigin` when going to `/data-unavailable`

```suggestion
   return redirectWithLocale(request, '/data-unavailable');
```

Reply:
@leiye29 corrected me. In this scenario, the `root.tsx` isn't hit and therefore the `userOriginCookie` will not be set at this point. Resolving.